src/Symfony/Component/HttpKernel/DataCollector/ConfigDataCollector.php

```diff
@@ -288,4 +285,108 @@ private function determineSymfonyState()
 
         return $versionState;
     }
+
+    /**
+     * Gets the names, paths and installed versions of the bundles enabled in
+     * the application.
+     *
+     * @return array
+     */
+    private function getBundleData()
+    {
+        $bundles = array();
+        $installedPackages = $this->getInstalledPackages();
+
+        foreach ($this->kernel->getBundles() as $name => $bundle) {
+            $bundlePath = $bundle->getPath();
+            $bundleVersion = null;
+
+            foreach ($installedPackages as $packageName => $packageVersion) {
+                if (strpos($bundlePath.'/', '/vendor/'.$packageName.'/')) {
+                    $bundleVersion = $packageVersion;
+
+                    break;
+                } elseif (strpos($bundlePath, 'symfony/symfony/src/Symfony/Bundle')) {
+                    // this is a built-in Symfony bundle; its version is the same as Symfony
+                    $bundleVersion = Kernel::VERSION;
+
+                    break;
+                }
+            }
+
+            $bundles[$name] = array('name' => $name, 'path' => $bundlePath, 'version' => $bundleVersion);
+        }
+
+        ksort($bundles);
+
+        return $bundles;
+    }
+
+    /**
+     * Finds the path of the 'composer.lock' file starting from the current
+     * kernel dir and going up recursively up until the project root dir.
+     *
+     * @return string
+     */
+    private function findComposerLockPath()
+    {
+        $path = $this->kernel->getRootDir();
+        $projectRootDirectory = $this->findProjectRootDirectory();
+
+        while ($path !== $projectRootDirectory) {
+            if (file_exists($composerLockPath = $path.'/composer.lock')) {
+                return $composerLockPath;
+            }
+
+            $path = realpath($path.'/..');
+        }
+    }
+
+    /**
+     * It returns the absolute path of the root directory of the Symfony project.
+     * This is usually the directory above the kernel directory, but in complex
+     * setups it can be different.
+     *
+     * @return string
+     */
+    private function findProjectRootDirectory()
+    {
+        // The project root directory is calculated as the longest prefix
+        // intersection between the kernel directory and the current file path
+        $paths = array($this->kernel->getRootDir(), __DIR__);
+
+        $longestCommonPrefix = array_reduce($paths, function ($prefix, $path) {
+            $length = min(strlen($prefix), strlen($path));
+            while (substr($prefix, 0, $length) !== substr($path, 0, $length)) {
+                --$length;
+            }
+
+            return substr($prefix, 0, $length);
+        });
+
+        return rtrim($longestCommonPrefix, '/');
+    }
+
+    /**
+     * Returns an array with the names and versions of the installed Composer
+     * packages for the current project.
+     *
+     * @return array
+     */
+    private function getInstalledPackages()
```

Comment:
Maybe the lock/json parts of composer can be removed from their main package so it can be reused? I'm not too familiar with the "behind the scenes" of the composer package itself, but there's a lot of small & reusable parts in there.